test/net/sourceforge/kolmafia/session/CrystalBallManagerTest.java

```diff
@@ -0,0 +1,67 @@
+package net.sourceforge.kolmafia.session;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.MonsterData;
+import net.sourceforge.kolmafia.combat.MonsterStatusTracker;
+import net.sourceforge.kolmafia.persistence.MonsterDatabase;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CrystalBallManagerTest {
+  static final MonsterData NAILER = MonsterDatabase.findMonster("smut orc nailer");
+  static final MonsterData SKELELTON = MonsterDatabase.findMonster("spiny skelelton");
+  static final MonsterData SKELTEON = MonsterDatabase.findMonster("party skelteon");
+
+  @BeforeAll
+  public static void beforeAll() {
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    KoLCharacter.reset("fakeUserName");
+    Preferences.reset("fakeUsername");
+    Preferences.setString(
+        "crystalBallPredictions",
+        "0:The Smut Orc Logging Camp:smut orc nailer|0:The Defiled Nook:party skelteon");
+    CrystalBallManager.reset();
+  }
+
+  @Test
+  public static void crystalBallZoneTest() {
+    assertTrue(CrystalBallManager.isCrystalBallZone("The Smut Orc Logging Camp"));
+    assertTrue(CrystalBallManager.isCrystalBallZone("The Defiled Nook"));
+    assertFalse(CrystalBallManager.isCrystalBallZone("The Defiled Niche"));
+  }
+
+  @Test
+  public static void crystalBallMonsterTest1() {
```

Comment:
Could you name these tests better? I'd like to be able to see the name of a test that has a failed and infer something from it